Route rule augmentation missing on 'nitropack/types' causes TS2353 in nuxt config

## 🐛 The bug

`@nuxtjs/robots` augments `NitroRouteRules` and `NitroRouteConfig` only on `declare module 'nitropack'`. The Nuxt config type system resolves route rules through `'nitropack/types'`, so declaring a `robots` rule in `routeRules` fails with TS2353.

The generated `.nuxt/types/nuxt-robots-nitro.d.ts` shows the gap.

```ts
declare module 'nitropack' {
  interface NitroRouteRules { robots?: RobotsValue | { indexable: boolean, rule: string } }
  interface NitroRouteConfig { robots?: RobotsValue | { indexable: boolean, rule: string } }
}
// No matching `declare module 'nitropack/types'` block.
```

Sibling Nuxt SEO modules (`@nuxtjs/sitemap`, `nuxt-og-image`, `nuxt-security`) augment both specifiers. Only robots is missing the second one.

## 🛠️ To reproduce

StackBlitz: https://stackblitz.com/github/JonathanXDR/repro-nuxtjs-robots-routerules

A minimal Nuxt 4 project. StackBlitz runs `npm run typecheck` on boot and produces:

```
nuxt.config.ts(10,7): error TS2353: Object literal may only specify known properties,
and 'robots' does not exist in type '{ cache?: ... }'.
      robots: false,
```

## 🌈 Expected behavior

`robots` should be a known property on `NitroRouteRules` and `NitroRouteConfig` when resolved through either `'nitropack'` or `'nitropack/types'`, matching how `@nuxtjs/sitemap`, `nuxt-og-image`, and `nuxt-security` already work. No shim should be needed in user projects.

Suggested fix: mirror the existing `declare module 'nitropack'` block with an identical `declare module 'nitropack/types'` block in the generated `.d.ts` (see `.nuxt/types/nuxt-sitemap-augments.d.ts` for the pattern).

## ℹ️ Additional context

|                      |                                                              |
| -------------------- | ------------------------------------------------------------ |
| **Operating system** | `macOS 25.5.0`                                               |
| **CPU**              | `Apple M4 Pro (14 cores)`                                    |
| **Node.js version**  | `v26.0.0`                                                    |
| **nuxt/cli version** | `3.35.2`                                                     |
| **Package manager**  | `npm@11.12.1`                                                |
| **Nuxt version**     | `4.4.6`                                                      |
| **Nitro version**    | `2.13.4`                                                     |
| **Builder**          | `vite@7.3.3`                                                 |
| **Config**           | `compatibilityDate`, `modules`, `routeRules`, `typescript`   |
| **Modules**          | `@nuxtjs/robots@6.0.8`                                       |

--- a/src/templates.ts
+++ b/src/templates.ts
@@ -1,6 +1,6 @@
 import type { Nuxt } from '@nuxt/schema'
 import type { ModuleOptions } from './module'
-import { addTypeTemplate } from '@nuxt/kit'
+import { addTypeTemplate, isNuxtMajorVersion } from '@nuxt/kit'
 
 interface TemplateContext {
   nuxt: Nuxt
@@ -8,12 +8,13 @@ interface TemplateContext {
 }
 
 export function registerTypeTemplates({ nuxt }: TemplateContext) {
-  // Nuxt 4 augments both 'nitropack' and 'nitropack/types' (matching @nuxt/nitro-server).
-  // Augmenting only one leaves the other's re-exported NitroRouteConfig/NitroRouteRules
-  // without our `robots` key, which breaks consumers like @nuxt/kit's extendRouteRules
-  // that pull the type through both module paths.
-  const isNuxt4 = Number(nuxt.options.future?.compatibilityVersion) === 4
-  const nitroModules = isNuxt4 ? ['nitropack', 'nitropack/types'] : ['nitropack']
+  // Nuxt 4 ships Nitro v3, which exposes `nitropack/types` as a public subpath.
+  // @nuxt/kit's extendRouteRules unions NitroRouteConfig across both 'nitropack'
+  // and 'nitropack/types', so augmenting only one leaves the other's re-exported
+  // interface without our `robots` key.
+  const nitroModules = isNuxtMajorVersion(4, nuxt)
+    ? ['nitropack', 'nitropack/types']
+    : ['nitropack']
 
   // Nuxt-only type augmentations (PageMeta)
   addTypeTemplate({

--- a/test/types/templates.test-d.ts
+++ b/test/types/templates.test-d.ts
@@ -7,6 +7,7 @@ import type {
 import type { H3EventContext } from 'h3'
 import type { NitroRouteConfig as NitroRouteConfigPack, NitroRouteRules as NitroRouteRulesPack } from 'nitropack'
 import type { NitroRouteConfig, NitroRouteRules, NitroRuntimeHooks } from 'nitropack/types'
+import type { NuxtConfig } from 'nuxt/schema'
 import type { PageMeta } from '#app'
 import { extendRouteRules } from '@nuxt/kit'
 import { describe, expectTypeOf, it } from 'vitest'
@@ -62,6 +63,19 @@ describe('nitropack augmentations (cross-module-path)', () => {
     }))
     expectTypeOf<NitroRouteConfig>().toExtend<{ ssr?: boolean, robots?: RobotsValue | { indexable: boolean, rule: string } }>()
     expectTypeOf<NitroRouteConfigPack>().toExtend<{ ssr?: boolean, robots?: RobotsValue | { indexable: boolean, rule: string } }>()
+  })
+})
+
+// Regression test for https://github.com/nuxt-modules/robots/issues/299
+// `nuxt.config.ts` routeRules are typed via `NuxtConfig['routeRules']`,
+// which (in Nuxt 4) resolves NitroRouteConfig through 'nitropack/types'.
+// If only 'nitropack' is augmented, declaring `robots` in routeRules fails
+// with TS2353 "Object literal may only specify known properties".
+describe('NuxtConfig routeRules (issue #299)', () => {
+  it('NuxtConfig.routeRules accepts a robots-tagged literal', () => {
+    expectTypeOf<{ '/admin': { robots: false } }>().toExtend<NonNullable<NuxtConfig['routeRules']>>()
+    expectTypeOf<{ '/private': { robots: 'noindex, nofollow' } }>().toExtend<NonNullable<NuxtConfig['routeRules']>>()
+    expectTypeOf<{ '/legacy': { robots: { indexable: false, rule: 'noindex' } } }>().toExtend<NonNullable<NuxtConfig['routeRules']>>()
   })
 })
 

